### PR TITLE
makefiles/buspirate: add Bus Pirate as Debug Adapter

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Bus Pirate is a multi-protocol analyzer that can be also used to generate signals.
It's pretty slow as it only supports software-bit banging for SWD, but if it's the only tool at hand it's better than nothing.

### Testing procedure

Flash any board with `PROGRAMMER=openocd DEBUG_ADAPTER=buspirate`

### CODE REVIEW

**Comment by maribu** [comment]
I just bought a Bus Pirate (v4). I expect the programmer to arrive in a month or so...

If the PR is not merged by then, I'll have a look.

**Comment by maribu** [comment]
On the other hand, I'm willing to ACK this PR untested. This is super un-scary. Mind to post output of `make flash` here?

**Comment by benpicco** [comment]
Unfortunately I don't have the Bus Pirate myself.
I was at a friend's place and we had to recover the bootloader on the [nRF52840 MDK USB Dongle](https://github.com/benpicco/RIOT/tree/boards/nrf52840-mdk-dongle).

The easiest way was to flash RIOT and then use the OpenOCD command line to flash the bootloader's hex file.
(Thinking about it, setting `FLASHFILE` on the command line should also work)

It's *really* slow, but worked reliable several times.

File: makefiles/tools/openocd-adapters/buspirate.inc.mk [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Bus Pirate debug adapter
+#
+# For SWD mode connect:
+#  MOSI - SWDIO
+#   CLK - SWCLK
+
+PROG_DEV ?= /dev/ttyUSB0
```
Comment (maribu) [inline makefiles/tools/openocd-adapters/buspirate.inc.mk:7]:
```suggestion

ifeq ($(OS),Darwin)
  PROG_DEV ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
else
  PROG_DEV ?= /dev/ttyUSB0
endif
```

**Pull request merged**